uamqp/aio/_cbs_async.py

```diff
@@ -0,0 +1,227 @@
+#-------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+#-------------------------------------------------------------------------
+
+import logging
+import asyncio
+from datetime import datetime
+
+from ._management_link_async import ManagementLink
+from ..utils import utc_now, utc_from_timestamp
+from ..message import Message, Properties
+from ..error import (
+    AuthenticationException,
+    TokenAuthFailure,
+    TokenExpired
+)
+from ..constants import (
+    CbsState,
+    CbsAuthState,
+    CBS_PUT_TOKEN,
+    CBS_EXPIRATION,
+    CBS_NAME,
+    CBS_TYPE,
+    CBS_OPERATION,
+    ManagementExecuteOperationResult,
+    ManagementOpenResult,
+    DEFAULT_AUTH_TIMEOUT
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def is_coroutine(get_token):
+    try:
+        if asyncio.iscoroutinefunction(get_token.func):
+            return True
+    except AttributeError:
+        if asyncio.iscoroutinefunction(get_token):
+            return True
+    raise ValueError("get_token must be a coroutine function")
```

Comment:
If we are enforcing that the token method be async - could we not simply `await` it an let the error raise if they provided a sync function? Rather than checking it?

Reply:
yep, this should be doable, I have removed the coroutine check